### PR TITLE
[Merged by Bors] - chore: Rename `Prod_map`

### PR DESCRIPTION
**Pull request by YaelDillies**
... in lemma names and section names in order to follow the naming convention.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — COMMENTED** [review]
This does seem better - but do we need aliases?

File: Mathlib/Data/Prod/Basic.lean [hunk 1]
```diff
@@ -19,9 +19,10 @@ It also defines better delaborators for product projections.
 variable {α : Type*} {β : Type*} {γ : Type*} {δ : Type*}
 
 @[simp]
-theorem Prod_map (f : α → γ) (g : β → δ) (p : α × β) : Prod.map f g p = (f p.1, g p.2) :=
-  rfl
-#align prod_map Prod_map
+theorem Prod.map_apply (f : α → γ) (g : β → δ) (p : α × β) : Prod.map f g p = (f p.1, g p.2) := rfl
```
Comment (urkud) [inline Mathlib/Data/Prod/Basic.lean:22]: IMHO, we should only simplify `Prod.map f g (x, y)` but this is a topic for another PR. Thanks! Please merge `master` to make sure that the CI is still happy, then merge.
bors d+

Comment (YaelDillies) [reply Mathlib/Data/Prod/Basic.lean:22]: I tend to agree but won't have time to run the experiment (nor do I need it)

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**